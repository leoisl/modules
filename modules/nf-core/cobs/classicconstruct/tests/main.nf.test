nextflow_process {
    name "Test Process COBS_CLASSICCONSTRUCT"
    script "../main.nf"
    process "COBS_CLASSICCONSTRUCT"

    tag "modules"
    tag "modules_nfcore"
    tag "cobs"
    tag "cobs/classicconstruct"

    test("sarscov2 - genome - fasta") {
        when {
            process {
                """
                input[0] = [
                    [ id:'test'], // meta map
                    file(params.test_data['sarscov2']['genome']['genome_fasta'], checkIfExists: true)
                    ]
                """
            }
        }
        then {
            assertAll(
                { assert process.success },
                { assert process.out.index != null },
                { assert process.out.index.get(0).get(1) ==~ ".*/index.cobs_classic" },
                { assert snapshot(process.out.index).match("index")},
                { assert snapshot(process.out.versions).match("versions") }
            )
        }
    }
}
