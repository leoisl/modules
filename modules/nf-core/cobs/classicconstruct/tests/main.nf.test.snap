{
    "versions": {
        "content": [
            [
                "versions.yml:md5,085a4365d164cf1a6d96c4deed57012a"
            ]
        ],
        "timestamp": "2023-12-06T18:53:19.374652"
    },
    "index": {
        "content": [
            [
                [
                    {
                        "id": "test"
                    },
                    "index.cobs_classic:md5,1745dca38a763f0b6a6e4943d53f8b6a"
                ]
            ]
        ],
        "timestamp": "2023-12-06T18:53:19.347747"
    }
}